speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/frontend/InterventionConfig.h

```diff
@@ -4,18 +4,27 @@
 #include "ConfigEnums.h"
 #include <unordered_map>
 
-// Location of the intervention XML file, with respect to the root data directory.
-#define INTERVENTION_DATA_DIR_FORMAT "%sdata/intervention/intervention.xml"
-#define SOUNDS_DIR_FORMAT "%sdata/intervention/sound/%s"
+// Location of the config.xml file with respect to the root data directory.
+#define CONFIG_XML_DIR_FORMAT "%sdata/intervention/config.xml"
+#define SOUNDS_DIR_FORMAT     "%sdata/intervention/sound/%s"
 
 // Parameters of sections and attributes used to search in the XML file.
 #define PRM_SECT_INTERVENTIONS  "Interventions"
+#define PRM_SECT_TEXTURE        "texture"
 #define PRM_SECT_SOUND          "sound"
-#define PRM_ATTR_NAME           "name"
-#define PRM_ATTR_TEXTURE        "texture"
+#define PRM_SECT_TEXT           "text"
+
 #define PRM_ATTR_XPOS           "xpos"
 #define PRM_ATTR_YPOS           "ypos"
 #define PRM_ATTR_SRC            "source"
+#define PRM_ATTR_CONTENT        "content"
+
+static std::unordered_map<InterventionAction, const char*> s_actionEnumParamMap = {
```

Comment:
shouldn't this be an array? the key is an integer, so indexing with it should be possible using the define statements